### DISCUSSION TITLE
inline grid adding of records

### DISCUSSION THREAD

**Discussion by shortsam1** [serenity-is/Serenity#5632] [Help] [answered]
I would like to know if its possible to produce a data entry form in the layout of a data grid with save and and new record options.  Something similar to the pic below but where every field is a data entry field or dropdown list with the option to save each line as its entered (save icon at the end of each line) and a add new icon below the last record entered, which will add a new line to the list. 

![image](https://user-images.githubusercontent.com/47765946/114832994-77b6ca00-9e12-11eb-9c56-433f9b986408.png)

Once the record is saved, it either disappears from the list or the list is cleared off on exiting the system and a new clean list is displayed on next login.

If this is possible, how do i go about this

**Comment by erehtolleh** [comment, marked as answer]
The buttons are already on top of the forms and does what you need.  Not sure why you need to change this  but I guess it's just a UI preference.

You can override the mylisthandler to not show save records  or in the xyzGrid.ts, create some filtering 
https://serenity.is/demo/BasicSamples/GridFilteredByCriteria
